libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts

```diff
@@ -631,6 +642,29 @@ export class SkyDateRangePickerComponent
       });
   }
 
+  #updateRequiredStates(): void {
```

Comment:
I agree, this solution feels a bit "hacky". 😅 Could we instead use Angular's `AbstractControl` API to respond to (and set) the required states for the child controls? I found a few examples online where folks were getting the `AbstractControl` of the parent `NgControl` and using that to set the required states. Here's a rough compilation of those findings that we might be able to use for this feature: https://stackblitz.com/edit/stackblitz-starters-javnqs?file=src%2Fcustom-form-control.ts

Reply:
This has given me some ideas. I'll work on tweaks.

Reply:
This is the article I robbed some stuff from: https://levelup.gitconnected.com/angular-get-control-in-controlvalueaccessor-b7f09a485fba

Reply:
I reworked this. NgControl is now pulled in during construction instead of being set in the validator. 

I also updated the shared form utility to be able to handle different form types and to not call the validator directly.